### ISSUE TITLE
FreeTube accessibility issues

### ISSUE THREAD

**Issue by RastislavKish** [FreeTubeApp/FreeTube#693]
Hello,
I was not sure whether to report this as a feature request or a bug, after some thinking, I've settled on the bug variant.
I'm a blind user of FreeTube, and while the app is definitely very interesting, there are some accessibility issues which would be very pleasant to have fixed.
They're mostly just playing with used HTML elements, so I hope it will be possible to implement them easily:

* Search results should be marked as headings. All that is needed to do here is to wrap the name of each result in perhaps h4 tag (as Search results heading uses h3), like Youtube does. Information about the author, what time ago was the video posted and the video's length can be included too, but in this case, it's important to arrange it in a way, that the name goes first and everything other follows it. Youtube has a good example of this. I don't know the visual specifics of FreeTube, so I'll leave upto authors how invasive would such rearrangement be.
* In settings, for some reason, the currently used combo-boxes are not accessible for screenreaders. Like, I see that there are combo-boxes, but I can't read the currently selected item and when i focus one and try to select something, my screenreader doesn't announce what I've selected. May be the items are not properly labelled? If you give me the used input type, I can write here an example of accessible combo-box HTML code.
* Screenreaders in general can't deal very well with clickable elements, like infact the all entries of the main menu (Trending, Playlists, Subscriptions, Settings), they're used also on video pages (to view comments, to view / hide replies to particular comment). It would be best to replace these with either links or buttons, which can be activated directly by pressing enter.
* On the page with a video, the video name should be marked as a heading too (as it leads the section with informations about video's author, likes, dislikes, description etc.).

That's what I've noticed so far, these four issues are from accessibility point of view the most important to fix, as they directly influence the app's usability for blind people.
The interface is otherwise very nice and hassle-free, that's why I hope it will be possible to do something about these things and make the app usable for everyone. :)

My configuration is Ubuntu mate 20.04 64-bit, screenreader orca v3.39.1 pre. It doesn't matter much I guess, as Electron's Chromium backend is the same for all platforms, adding it just for completeness.
FreeTube version 0.9.0.

Thanks for the great project!

**Comment by peepopoggers** [comment]
@RastislavKish do you have problems interpreting icons only (compared to with text)? thanks.

**Comment by RastislavKish** [comment]
Some accessibility updates:
* The settings page seems to work now! I don't know the exact version when this happened as I usually don't need to tweak them very often, but the comboboxes seem usable now.
* Also, the new menu design from 0.13.0 makes it nicely usable due to the usage of buttons. There are still many "fake controls" in the app, but this is anyway a great accessibility improvement.

@peepopogger currently, the most notable issue are the "fake controls". For example, when a paragraph (<p>) is given a text of a button and it gets a click listener, it will in theory behave like a button, but it's not really one (paragraphs are intended only for displaying data like text, they are not intended to be used as controls (like buttons, links, etc.)).

For a sighted person, the difference is unnoticeable as they click on things anyway, so it doesn't really matter what did they click on.

But screenreaders have a problem, because paragraphs don't react to key presses (like return), for example, even in your browser, you can tab to a button and activate it with the return key or spacebar, but you can't do this with clickable paragraphs (like the one for viewing comments in FreeTube).

**Comment by jasonhenriquez** [comment]
I'm working on replacing the many "fake controls" with buttons and links now. @RastislavKish Sorry this has taken so long to get addressed. I would consider this accessibility issue to be a severe bug.

**Comment by jasonhenriquez** [comment]
Is it acceptable for me to link my pull request to this issue, or are any of your checkboxes still not sufficiently addressed in your opinion? @RastislavKish

**Comment by RastislavKish** [comment]
@jasonhenriquez sure, your pr has resolved all the issues, so I think linking it here would be more than appropriate. :)

Thank you again very much for all the great work on this, I've already updated the checkboxes.

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
@ChunkyProgrammer Its been a while can this be closed?

**Comment by jasonhenriquez** [comment]
I'd still like to hear if @RastislavKish and/or any other disabled peeps have any more areas for improvement to share before closing this out.

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Maybe this can be enhanced, 
- Users are unable to tab to the search icon to press enter and search (same case for creator channel page)
- Users are unable to tab to the arrow in the blacklist to press enter to blacklist something (same case in external player settings)

https://github.com/FreeTubeApp/FreeTube/assets/73130443/f47fcc9e-c1bc-403c-940f-2f50a7545674

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
> I hate to ask, but for Blind Users, is there going to be an option to disable to VIDEO portion? I'd like to stream just the AUDIO (to save bandwidth).
> 
> I have wanted that feature out of YouTube for decades (along with better playlist management / sharing).
> 
> I have low-bandwidth and a governor. When I use too much data, my internet slows to a crawl.

@aaronsql2019 you can already do this in 2 ways: 

1. Every video plays Audio format by default: Settings -> Player settings -> Default Video Format -> Audio Formats
2. Selectively change the currently playing video to play only audio format: Under the currently playing video to the left of the share button there is a button to change the format of the video

**Comment by aaronsql2019** [comment]
THANK YOU SO MUCH. I've been looking for that feature for so long. I'm SOOOOOOOOOOOOOOOOOOO glad you support it.

Now, if you could go into Spotify and allow me to 'ENABLE VIDEOS' I'd be as happy as a clam. 
LOL. I'm only kidding.

I've been waffling about dedicating myself to spotify or youtube. I can't make sense of YouTube playlists.
I fork a playlist. I want to merely REMOVE a video. I can't for the life of me figure it out.

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
@ozrendev is this maybe something you would like to pick up

https://github.com/FreeTubeApp/FreeTube/issues/693#issuecomment-1824975963

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
@RastislavKish do you have more improvement suggestions?
